File: frontend/src/app/services/models/profile-dto.ts
/* tslint:disable */
/* eslint-disable */
import { Attachment } from '../models/attachment';
export interface ProfileDto {
  followersCount?: number;
  id?: number;
  profilePicture?: Attachment;
  subscribed?: boolean;
  subscriptionsCount?: number;
  username?: string;
}

File: src/main/java/com/javadevmz/my_social_media/dao/dto/ProfileDto.java
package com.javadevmz.my_social_media.dao.dto;

import com.javadevmz.my_social_media.dao.entity.Attachment;
import com.javadevmz.my_social_media.dao.entity.User;

import java.time.LocalDateTime;

public record ProfileDto(Long id, String username, Attachment profilePicture, Boolean subscribed, Long followersCount, Long subscriptionsCount) {
}
